lib/cadet/assessments/assessments.ex

```diff
@@ -488,16 +488,17 @@ defmodule Cadet.Assessments do
       |> where(submission_id: ^id)
       |> join(:inner, [a], q in assoc(a, :question))
       |> join(:left, [a, _], g in assoc(a, :grader))
-      |> preload([_, q, g], question: q, grader: g)
+      |> join(:inner, [a, _, _], s in Submission, a.submission_id == s.id)
+      |> join(:inner, [a, _, _, s], st in User, s.student_id == st.id)
+      |> preload([_, q, g, s, st], question: q, grader: g, submission: {s, student: st})
 
     cond do
       role in @grading_roles ->
         students = Cadet.Accounts.Query.students_of(grader)
 
         answers =
           answer_query
-          |> join(:inner, [a], s in Submission, a.submission_id == s.id)
-          |> join(:inner, [a, _, _, s], t in subquery(students), t.id == s.student_id)
+          |> join(:inner, [_, _, _, s, _], t in subquery(students), t.id == s.student_id)
```

Comment:
Prefer `|> join(:inner, [..., s, _], t in subquery(students), t.id == s.student_id)`